contracts/CMakeLists.txt

```diff
@@ -2,3 +2,5 @@ add_subdirectory(eden)
 add_subdirectory(token)
 add_subdirectory(boot)
 add_subdirectory(bios)
+add_subdirectory(activate_feature)
+add_subdirectory(set_parameters)
```

Comment:
Why use new contracts rather than boot/bios which serve a similar purpose.  Switching out the system contract behind the scenes seems like a good way to cause surprises.

Reply:
`boot` has a fixed config, suitable for Eden testing. I wanted a more modular system that doesn't leave a bios contract installed, unless the user wants a bios contract.

Reply:
The whole point of the bios contract is to allow access to configuration like `set_parameters`.  The only real reason not to want the bios contract is if you're installing your own system contract.  The point of the boot contract (at least, the original point--we've shoved more into in since) was to make the bios loadable.

Reply:
Over time, the bios contract would depend on more HFs being activated, making it an all-or-nothing, unless clsdk builds the bios contract while building the user's test suite.

Reply:
That's true, but
- Not all protocol features need configuration
- Very few contract developers will care if they get a few extra protocol features.
- `preactivate_feature` is must-have which the tester activates unconditionally and `set_blockchain_parameters_packed` is baseline.

If we do create separate contracts I would strongly prefer to create additional accounts for them instead of blanking out eosio every time the user runs some configuration.

Reply:
Additional accounts adds the `set_privileged` headache. We can clean up `boot` and `bios` in the future, giving test authors a choice between the two approaches.

Reply:
Why would users want a choice?  They won't understand what they're choosing.  They might even try to use both in the same test, which would be bad.

I think the side-effect of clearing code on eosio when setting configuration is unacceptable.  I'll take pretty much any other approach over that.

Reply:
I agree with Steven

Reply:
A few ideas:
- Reset code to its original value.  Side-effect: increments code sequence.  I think this is still ugly, but I can live with it.
- Fail if eosio has a contract.  This doesn't prevent errors, but it will tell users what they did wrong.
- Bios + protocol feature specific contracts.  Precondition: eosio must hold a compatible contract, with setpriv at a minimum.  Side-effect: extra accounts created.
- bios provides all functionality.  Some protocol features must be activated during setup, whether users want them or not.
- bios provides base functionality and handles protocol feature specific behavior either by modifying itself or deploying additional contracts on demand.  The implementation is painful, but the user-visible behavior is easy to describe.